src/cargo/util/toml_mut/manifest.rs

```diff
@@ -363,6 +363,30 @@ impl LocalManifest {
         Ok(())
     }
 
+    /// Add feature entry to a Cargo.toml.
+    pub fn insert_into_feature_table(
+        &mut self,
+        table_path: &[String],
+        dep: &Dependency,
+    ) -> CargoResult<()> {
```

Comment:
This function has a fairly specific policy (handling for this feature) where the name is fairly generic.

Reply:
FYI the name is still fairly generic

Reply:
what about `explicit_activate_optional_dependency_as_feature`